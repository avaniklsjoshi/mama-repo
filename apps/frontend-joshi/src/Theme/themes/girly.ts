export default {
  themeType: "girly",
  colors: {
    background: "#f1b7ea",
    text: "#fff"
  },
  header: {
    background: "#000",
    text: "#fff"
  },
  footer: {
    background: "#262726",
    text: "#fff"
  },
  section: {
    background: "#008F7A",
    text: "#fff"
  }
};
