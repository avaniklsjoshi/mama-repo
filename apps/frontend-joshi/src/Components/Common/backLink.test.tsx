/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
import { act, fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import BackLink from "./backLink";
import girly from "../../Theme/themes/girly";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: any) => key })
}));

describe("BackLink", () => {
  it("renders BackLink", () => {
    const { container } = render(
      <ThemeProvider theme={girly}>
        <Router>
          <BackLink />
        </Router>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
    const featureButton = screen.getByTestId("back-link");
    fireEvent.click(featureButton);

    expect(screen.getByText("back")).toBeInTheDocument();
  });
});
