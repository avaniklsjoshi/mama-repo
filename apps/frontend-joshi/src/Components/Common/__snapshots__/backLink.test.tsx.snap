// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackLink renders BackLink 1`] = `
<div>
  <div
    class="sc-bczRLJ fQltev"
  >
    <a
      class="back-link"
      data-testid="back-link"
      style="color: red;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ArrowBackIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      back
    </a>
  </div>
</div>
`;
